### PR TITLE
Edit Stage Workflow

### LINKED ISSUE #55: Create Edit Stage Modal

### LINKED ISSUE #88: Generate helper function for Edit Stage for OpenAI

### LINKED ISSUE #138: Edit Stage Backend

### PR DESCRIPTION
**Pull request by L0Lmaker**
Try editing a stage from the see more panel. Let me know if you have any concerns.

### CODE REVIEW

File: client/src/components/TripElement/EditStageForm.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  return (
+    <Modal
+      open={editStageModalIsOpen}
+      handleClose={handleCloseEditStageModal}
+      title='Adjusting Course...'
```
Comment (vee-16) [inline client/src/components/TripElement/EditStageForm.js]:
```suggestion
      title='Adjusting Plan...'
```

**Comment by vee-16** [comment]
should we add an alert showing what got updated?

**Comment by vee-16** [comment]
Updated form titles for edit trip and edit stage:
- Adjust Course changed to Adjust Trip and Adjust Plan

Added minor fixes to:
- generateStage prompt to enforce preferences, and to get places identified by google maps
- passing `newStageResponse.newStage.stageLocation` in StageController to improve response sent to google api for fetching coordinates (was having undefined issues with stageLocation previously with some edit stage)

File: client/src/components/TripElement/StageForm.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        <div className={'text-2xl'}>
+          {stage.emoji} {stage.stageLocation}
+        </div>
+        <div>{stage.description}</div>
+      </div>
+      <TextField
+        {...register('updateNotes', { required: false })}
+        label='Update'
+        placeholder='Tell me what you would like to change about this...'
+      />
```
Comment (NLmeng) [inline client/src/components/TripElement/StageForm.js]: it works, just wondering if we should allow update with no prompt

File: client/src/components/TripElement/EditStageForm.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  return (
+    <Modal
+      open={editStageModalIsOpen}
+      handleClose={handleCloseEditStageModal}
+      title='Adjusting Plan...'
```
Comment (NLmeng) [inline client/src/components/TripElement/EditStageForm.js]: would be very cool if you could use the same animated title like the other modals

**Review by NLmeng — APPROVED** [review]
update root readme and cross `Users can manually update an individual destination within a plan.`

File: client/src/components/TripElement/StageForm.js (hunk 2 above)
Comment (L0Lmaker) [reply client/src/components/TripElement/StageForm.js]: good point, ill make it required

File: client/src/components/TripElement/StageForm.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      className='flex flex-col space-y-4 text-center'
+    >
+      <div>
+        <div className={'text-2xl'}>
+          {stage.emoji} {stage.stageLocation}
+        </div>
+        <div>{stage.description}</div>
+      </div>
+      <TextField
+        {...register('updateNotes', { required: false })}
```
Comment (L0Lmaker) [inline client/src/components/TripElement/StageForm.js]:
```suggestion
        {...register('updateNotes', { required: true })}
```

**Pull request merged**